datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -784,6 +790,29 @@ fn get_corrected_filter_mask(
             corrected_mask.extend(vec![Some(false); null_matched]);
             Some(corrected_mask.finish())
         }
+        JoinType::LeftMark => {
```

Comment:
this block is for filtered joins, does LeftMark need it?

Reply:
I think it needed for `LeftMark` as well. Currently it might only be tested by fuzz test, but I think this is reachable from SQL as well (using subquries).